code/modules/power/singularity/singularity.dm

```diff
@@ -129,21 +132,23 @@
 		if(3)
 			energy -= round(((energy+1)/4),1)
 
-/obj/singularity/process()
+/obj/singularity/process(delta_time)
 	if(current_size >= STAGE_TWO)
 		if(prob(event_chance))//Chance for it to run a special event TODO:Come up with one or two more that fit
 			event()
-	dissipate()
+	dissipate(delta_time)
 	check_energy()
 
-/obj/singularity/proc/dissipate()
-	if(!dissipate)
+/obj/singularity/proc/dissipate(delta_time)
+	if (!dissipate)
 		return
-	if(dissipate_track >= dissipate_delay)
-		src.energy -= dissipate_strength
-		dissipate_track = 0
-	else
-		dissipate_track++
+
+	time_since_last_dissipiation += delta_time
+
+	// Uses a while in case of especially long delta times
+	while (time_since_last_dissipiation >= dissipate_delay)
+		energy -= dissipate_strength
```

Comment:
Couldn't you just do `energy -= strength * delta_time`, then you dont need the extra variable nor the while loop

Reply:
I'm not sure what you mean by this comment--this would be completely different behavior instead of the existing one I am trying to emulate.

Reply:
It would still have the same average energy loss per second, making the dissipation more smooth. But as you say, maybe this isn't what we want it to do.

Reply:
Looking over the rest of the code, I don't see any problem with the energy not having these large discrete steps

Reply:
I don't have a large opinion of it one way or the other, just sticking with the original design of the singularity which solved it this way for some reason unbeknownst to me. If you would like to make your own PR after the fact, feel free to do so, but this is meant to just to maintain parity.